tests/integration/web/maintenance/views_test.py

```diff
@@ -152,6 +152,80 @@ def test_when_existing_task_is_requested_it_should_render_with_intact_descriptio
         assert empty_maintenance_task.description in smart_str(response.content)
 
 
+class TestSearchMaintenanceComponents:
+    def test_component_search_returns_results(self, db, client):
+        url = reverse('maintenance-component-search')
+        response = client.post(url, {'search': 'test-gsw'})
```

Comment:
Funnily enough, I've forgotten all about the test devices we add by default into our our test data using pure SQL :laughing: 

Most of our modern tests actually employ pytest fixtures to insert test data as needed, but this is OK, I guess :-)